### PR TITLE
DOCS-213 Clarify Differences in Persistence

### PR DESCRIPTION
**Pull request by angelasimms**
Draft for review. Just wanted to check that I've understood the problem correctly and got the right approach. If all is OK I'll get a technical review.

### CODE REVIEW

**Review by JakeSCahill — APPROVED** [review]
I think this resolves the issue nicely.

File: docs/modules/storage/pages/high-density-memory.adoc [hunk 1]
```diff
@@ -245,120 +249,85 @@ PersistentMemoryConfig pmemConfig = memoryConfig.getPersistentMemoryConfig()
...
-** `SYSTEM_MEMORY`: If you choose this mode, the persistent memory is onlined as system memory (aka KMEM DAX).
-* `directories`: List of the persistent memory mounting directories to store data of all the data structures backed by
-High-Density Memory Store. When you provide the directories:
-** the persistent memory usage is enabled automatically and you do not need to explicitly
+The following elements and attributes are also used in the configuration: 
+
+* `enabled`: Specifies whether use of memory on the persistent memory device is enabled or not. The default value is `false`, that is use of the device is disabled.
+* `mode`: Defines the operational mode of the persistent memory device. Two modes are supported:
+** `MOUNTED`: If you choose this mode, the memory is mounted into the file system (aka FS DAX).
+** `SYSTEM_MEMORY`: If you choose this mode, the memory is onlined as system memory (aka KMEM DAX).
```
Comment (blazember) [inline docs/modules/storage/pages/high-density-memory.adoc:257]: Actually, the memory has to be already online. The onlining is done at system level, this option just makes HZ using that portion of the memory.

**Review by blazember — APPROVED** [review]
Approving with some minors to consider.

File: docs/modules/storage/pages/high-density-memory.adoc [hunk 2]
```diff
@@ -159,25 +160,28 @@ See the xref:ROOT:system-properties.adoc[System Properties appendix] on how to u
...
-The optional `persistent-memory` element in the `native-memory` configuration block
-enables the persistent memory usage and defines the directories where this memory
-is mounted along with its operational mode. See the element descriptions below the
-following configuration snippets.
+=== Configuring Intel(R) Optane(TM)
+
+Prerequisites:
+
+- Linux x86_64 operating system
+- Dual socket machine with both sockets populated with Intel(R) Optane(TM) DC DIMMs configured in interleaved mode
```
[the remark this answers is not in the document]
Comment (angelasimms) [reply docs/modules/storage/pages/high-density-memory.adoc:178]: I've added a note to try to make this clearer.

File: docs/modules/storage/pages/high-density-memory.adoc [hunk 3]
```diff
@@ -159,25 +160,28 @@ See the xref:ROOT:system-properties.adoc[System Properties appendix] on how to u
...
+- DIMMs mounted as  `/mnt/pmem0` and `/mnt/pmem1`, and known as NUMA node0 and node1 respectively
+
+NOTE: Currently the only supported configuration is Intel(R) Optane(TM) DC DIMMs running on Linux x86_64.
```
[the remark this answers is not in the document]
Comment (angelasimms) [reply docs/modules/storage/pages/high-density-memory.adoc]: Updated as requested.

File: docs/modules/storage/pages/high-density-memory.adoc (hunk 1 above)
Comment (angelasimms) [reply docs/modules/storage/pages/high-density-memory.adoc]: I've updated the description but could you have a quick check to make sure that this is still accurate?

Comment (blazember) [reply docs/modules/storage/pages/high-density-memory.adoc]: Maybe "...Hazelcast uses the already onlined PMEM portion of the system memory..." indicates it better that the user has something to do in order to use this mode.

Comment (angelasimms) [reply docs/modules/storage/pages/high-density-memory.adoc]: Updated as requested.

**Pull request merged**